akka-stream/src/main/scala/akka/stream/impl/StreamLayout.scala

```diff
@@ -311,7 +311,16 @@ private[stream] class VirtualSubscriber[T](val owner: VirtualPublisher[T]) exten
  */
 private[stream] class VirtualPublisher[T]() extends Publisher[T] {
   @volatile var realPublisher: Publisher[T] = null
-  override def subscribe(s: Subscriber[_ >: T]): Unit = realPublisher.subscribe(s)
+  override def subscribe(s: Subscriber[_ >: T]): Unit = {
+    val sub = realPublisher.subscribe(s)
+    realPublisher match {
+      case _: SupportingMultipleSubscribers   ⇒
```

Comment:
ok, I'll take another look at what happens when I simply remove the `SupportingMultipleSubscribers` marker, but I think I added it for the reason that we have tests that will fail otherwise. E.g. FuturePublisher, EmptyPublisher and SynchronousIterablePublisher have native support for more than one subscriber.